src/pages/product-list-page/product-list-filter/product-list-filter.js

```diff
@@ -0,0 +1,328 @@
+
+  const colors = [
+    ...new Set(
+      products.map((product) => JSON.stringify(product.colors[0].simpleName))
+    )
+  ].map(JSON.parse);
+  const categories = [
+    ...new Set(products.map((product) => JSON.stringify(product.category)))
+  ].map(JSON.parse);
+
+  const patterns = [
+    ...new Set(
+      products.map((product) =>
+        product.pattern.length
+          ? JSON.stringify(product.pattern)
+          : JSON.stringify([
+            {
+              lang: 'uk',
+              value: 'Немає'
+            },
+            {
+              lang: 'eng',
+              value: 'None '
+            }
+          ])
+      )
+    )
+  ].map(JSON.parse);
+  const [search, setSearch] = useState('');
+  const [price, setPrice] = useState([null, null]);
+  const [categoryCheck, setCategoryCheck] = useState({});
+  const [colorsCheck, setColorsCheck] = useState({});
+  const [patternsCheck, setPatternsCheck] = useState({});
+
+  const handleSearch = (event) => {
+    setSearch(event.target.value);
+  };
+
+  const handlePriceChange = (event, newValue) => {
+    setPrice(newValue);
+  };
+
+  const handleCategoryChange = (event) => {
+    setCategoryCheck({
+      ...categoryCheck,
+      [event.target.name]: event.target.checked
+    });
+  };
+
+  const handleColorChange = (event) => {
+    setColorsCheck({
+      ...colorsCheck,
+      [event.target.name]: event.target.checked
+    });
+  };
+
+  const handlePatternChange = (event) => {
+    setPatternsCheck({
+      ...patternsCheck,
+      [event.target.name]: event.target.checked
+    });
+  };
+
+  const handleFilter = () => {
+    dispatch(
+      setColorsFilter(
+        colors
+          .filter((color) => colorsCheck[color[1].value])
+          .map((color) => color[1].value)
+      )
+    );
+    dispatch(
+      setPatternsFilter(
+        patterns
+          .filter((pattern) => patternsCheck[pattern[1].value])
+          .map((pattern) => pattern[1].value)
+      )
+    );
+    dispatch(
+      setCategoryFilter(
+        categories
+          .filter((category) => categoryCheck[category.name[1].value])
+          .map((category) => category._id)
+      )
+    );
+    dispatch(setSearchFilter(search));
+    dispatch(setPriceFilter(price));
+    dispatch(
+      getFiltredProducts({
+        search,
+        price,
+        colors: colors
+          .filter((color) => colorsCheck[color[1].value])
+          .map((color) => color[1].value),
+        patterns: patterns
+          .filter((pattern) => patternsCheck[pattern[1].value])
+          .map((pattern) => pattern[1].value),
+        category: categories
+          .filter((category) => categoryCheck[category.name[1].value])
+          .map((category) => category._id),
+        skip: currentPage * productsPerPage,
+        limit: productsPerPage,
+        basePrice: sortByPrice,
+        rate: sortByRate,
+        purchasedProducts: sortByPopularity
+      })
+    );
+  };
+
+  const handleClearFilter = () => {
```

Comment:
`handleClearFIlter` should be an action that you dispatch and inside the store, you reset state of filter to the default state

Reply:
fixed